bigbone/src/main/kotlin/social/bigbone/api/method/OAuthMethods.kt

```diff
@@ -24,18 +24,49 @@ class OAuthMethods(private val client: MastodonClient) {
      *  Must match one of the redirect_uris declared during app registration.
      * @see <a href="https://docs.joinmastodon.org/methods/oauth/#authorize">Mastodon oauth API methods #authorize</a>
      */
+    @JvmOverloads
     fun getOAuthUrl(clientId: String, scope: Scope, redirectUri: String = "urn:ietf:wg:oauth:2.0:oob"): String {
         val endpoint = "oauth/authorize"
         val params = Parameters()
             .append("client_id", clientId)
             .append("redirect_uri", redirectUri)
             .append("response_type", "code")
             .append("scope", scope.toString())
-        return MastodonClient.fullUrl(client.getInstanceName(), endpoint, params).toString()
+        return MastodonClient.fullUrl(client.getScheme(), client.getInstanceName(), client.getPort(), endpoint, params).toString()
     }
 
     /**
-     * Obtain an access token using OAuth 2 authorization code grant type. To be used during API calls that are not public.
+     * Obtain a client access token using OAuth2 client credentials grant type.
+     * @param clientId The client ID, obtained during app registration.
+     * @param clientSecret The client secret, obtained during app registration.
+     * @param scope List of requested OAuth scopes, separated by spaces. Must be a subset of scopes declared during app registration.
+     * @param redirectUri Set a URI to redirect the user to. Defaults to "urn:ietf:wg:oauth:2.0:oob",
+     *  which will display the authorization code to the user instead of redirecting to a web page.
+     *  Must match one of the redirect_uris declared during app registration.
+     *  @see <a href="https://docs.joinmastodon.org/methods/oauth/#token">Mastodon oauth API methods #token</a>
+     */
+    @JvmOverloads
+    fun getClientAccessToken(
```

Comment:
I will undo this implementation of OAuth2 password grant flow, as you have already implemented it in https://github.com/andregasser/bigbone/pull/147.

Reply:
Done.